### PR TITLE
feat(cli): option to open the project in the browser

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Opens the server listeners url in the default browser

```bash
nuxt -o
//or
nuxt --open
````

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by pi0** [comment]
`open` package size seems good: [install size](https://packagephobia.now.sh/result?p=open)

File: packages/cli/src/commands/dev.js [hunk 1]
```diff
@@ -9,17 +10,29 @@ export default {
...
+      for (const listener of nuxt.server.listeners) {
+        await opn(listener.url)
+      }
+    }
   },
 
   async startDev(cmd, argv) {
     try {
-      await this._startDev(cmd, argv)
+      return await this._startDev(cmd, argv)
```
Comment (pi0) [inline packages/cli/src/commands/dev.js]: Please use named return like `const nuxt = await` + `return { nuxt }`

Comment (ricardogobbosouza) [reply packages/cli/src/commands/dev.js]: @pi0 renamed

**Review by pi0 — DISMISSED** [review]

**Comment by ricardogobbosouza** [comment]
@pio `opn` package was the best and most used I found

**Comment by pi0** [comment]
@ricardogobbosouza Yeah seems light-weight. The only problem I can predict is that we have the plan to try out zero dependency builds of nuxt packages. Packages that have binary files inside can be little problematic as it can't be bundled. However, my comment was general. No worries until then.

**Review by kevinmarrec — DISMISSED** [review]

**Comment by kevinmarrec** [comment]
Nice ! I'm wondering why [webpack-bundle-analyzer](https://www.npmjs.com/package/webpack-bundle-analyzer) (that we use for the `--analyze` option of the `build`command) is using [opener](https://www.npmjs.com/package/opener) instead of [opn](https://www.npmjs.com/package/opn)

**Comment by pi0** [comment]
[opener code](https://unpkg.com/opener@1.5.1/lib/opener.js) seems has an explicit workaround for WSL and also has no binary inside it. So maybe worth comparing them?

**Comment by ricardogobbosouza** [comment]
In my opinion I prefer the `opener` because it is smaller and does not contain a binary inside it.

**Review by galvez — DISMISSED** [review]

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]
Evaluate `opn` vs `opener` first.

**Comment by ricardogobbosouza** [comment]
Any tips on what you guys use to compare?

**Comment by manniL** [comment]
@ricardogobbosouza 

Some criterias
* Bundle Size
* System compatibility (OSX, Win, *NIX) / Interoperability
* Stability
* "Burdens" (like extra binaries)

**Comment by ricardogobbosouza** [comment]
@manniL a quick comparison

https://bundlephobia.com/result?p=opn@5.4.0
https://bundlephobia.com/result?p=opener@1.5.1

`opener` is almost half the size, with no dependencies and no binaries

**Comment by ricardogobbosouza** [comment]
I will change to the `opener`

**Comment by pimlie** [comment]
One of the reasons opener is smaller is because opn bundles the xdg-open sh script itself. I guess thats to not to rely on the local system to have (an updated version of) xdg-open installed.

Looking at the wsl detection opn (or more specifically is-wsl) has an extra check. Unfortunately the extra dependency cannot really be used as an argument, because sindresorhus always splits his code in tiny packages and have them as deps. He claims that that increases his productiveness.

Although I am not sure what the usefulness is for us, the opn package seems to have a bit more features with the trade-off that its a bigger install size.

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
We want something small that just works. I suggest let's try `opener`. If we found any unusual problems with it fork and improve or change to another package to prevent reinventing.

**Comment by Atinux** [comment]
Actually, I believe this open should be by default on `nuxt dev` for Nuxt 3 when we will have:
- Module install with the command line = Nani?
- HMR for server parts (server middleware, etc) = Nuxt workers
- Nuxt UI that show the build progress (instead of a pending page waiting for server resources)

I am open to have it as an option at the moment, thank you @ricardogobbosouza for this PR <3

**Comment by ricardogobbosouza** [comment]
@Atinux We can add as default in Nuxt 3.
Most frameworks come as standard, eg: Ionic

**Comment by pimlie** [comment]
Please also keep in mind that not everybody develops on localhost. We could check for e.g. `SSH_TTY` env variable to check if somebody is logged in from remote and then not open the url by default (Because otherwise you have to close Links or lynx everytime you run nuxt dev and nobody develops their website for text-mode anyway)

**Comment by Atinux** [comment]
Really good remark @pimlie

**Review by TheAlexLichter — APPROVED** [review]

**Pull request merged**